### PR TITLE
[SW-2514] Delete Train and Validation H2O Frame after Training a Model

### PR DESCRIPTION
**Pull request by mn-mikke**
The H2O Frames are created during the call of SW wrapper's `fit` function. The frames are accessible just via Flow UI and are not exposed to the user. We can consider this as a memory leak.

### CODE REVIEW

**Review by honzasterba — APPROVED** [review]

**Comment by honzasterba** [comment]
what about some tests making sure that generated objects are being cleaned up?

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala [hunk 1]
```diff
@@ -90,6 +90,8 @@ class H2OAutoML(override val uid: String)
 
     val algoName = getLeaderboard().select("model_id").head().getString(0)
     deleteRegisteredH2OFrames()
+    train.delete()
```
Comment (michalkurka) [inline ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala:93]: should you instead call registerFrameForDeletion? and let the frames be cleaned up by deleteRegisteredH2OFrames?

**Comment by michalkurka** [comment]
unrelated idea: I checked the method "prepareDatasetForFitting" and got the idea of converting training and validation frame at the same time in parallel

question: didn't notice blending frame - do we support blending in SW?

**Comment by mn-mikke** [comment]
> question: didn't notice blending frame - do we support blending in SW?
@michalkurka  `blending_frame` and `leaderboard_frame" haven't exposed on AutoML yet. If these frames are not too big (can be serialized to one node), we can expose them. We already have parameters of the `DataFrame` type in SW.

**Comment by michalkurka** [comment]
@mn-mikke Leaderboard is tiny

blending frame is just like validation frame (same size), it is used on AutoML input when people don't want to do cross validation

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala (hunk 1 above)
Comment (mn-mikke) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala:93]: Like this idea!

**Review by honzasterba — APPROVED** [review]

**Review by Pscheidl — APPROVED** [review]

**Pull request merged**